casper/src/main/scala/coop/rchain/casper/MultiParentCasperImpl.scala

```diff
@@ -108,6 +108,29 @@ class MultiParentCasperImpl[F[_]: Sync: Capture: ConnectionsCell: TransportLayer
                }
     } yield result
 
+  def refetchDependencies(b: BlockMessage): F[Unit] =
+    for {
+      dag            <- blockDag
+      missingParents = parentHashes(b).toSet
+      missingJustifications = b.justifications
+        .map(_.latestBlockHash)
+        .toSet
+      allDependencies = (missingParents union missingJustifications).toList
+      missingDependencies = allDependencies.filterNot(
+        blockHash =>
+          dag.dataLookup.contains(blockHash) || blockBuffer.exists(_.blockHash == blockHash)
+      )
+      _ <- missingDependencies.traverse(hash => handleMissingDependency(hash, b))
+      possiblyFailedDependencyFetches = allDependencies.foldLeft(Set.empty[BlockMessage]) {
+        case (acc, blockHash) =>
+          blockBuffer.find(_.blockHash == blockHash) match {
+            case Some(block) => acc + block
+            case None        => acc
+          }
+      }
```

Comment:
The computation of possiblyFailedDependencyFetches can be written in a more readable fashion
```      possiblyFailedDependencyFetches = 
        allDependencies.flatMap(blockHash => blockBuffer.find(_.blockHash == blockHash))
          .foldLeft(Set.empty[BlockMessage]) {case (acc,blockMessage) => acc + blockMessage}
```